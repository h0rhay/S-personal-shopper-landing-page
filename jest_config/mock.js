export const mockTheme = {
  themeType: 'dark',
  breakpoints: {
    small: '(max-width: 812px)',
    medium: '(min-width: 813px)',
    large: '(min-width: 960px)',
    deviceSwitchSm: '(max-width: 896px)',
    deviceSwitchLg: '(min-width: 897px)',
  },
  typography: {
    mobile: {
      headerOne: {
        fontSize: '3rem',
      },
      headerTwo: {
        fontSize: '1.75rem',
      },
      displayHeading: {
        fontSize: 'calc(1.5rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
    desktop: {
      headerOne: {
        fontSize: '4rem',
      },
      headerTwo: {
        fontSize: '2rem',
      },
      displayHeading: {
        fontSize: 'calc(4rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
  },
  text: '#ffffff',
  primary: '#212121',
  primaryAccent: '#2E2E2E',
  accent: '#ffe256',
  shadow: 'rgba(0, 0, 0, 0.4)',
};
