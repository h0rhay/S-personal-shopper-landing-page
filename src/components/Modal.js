import PropTypes from 'prop-types';
import React from 'react';
import { DialogContent, DialogOverlay } from '@reach/dialog';
import { useTransition, animated } from 'react-spring';
import styled, { css } from 'styled-components';
import '@reach/dialog/styles.css';
import CloseSVG from '../images/cross-icon.svg';

const DialogOverlayComponent = styled(DialogOverlay)`
  background: hsla(0, 0%, 0%, 0.7);
  z-index: 10;
`;

const DialogContentComponent = styled(DialogContent)`
  margin: 30vh auto;
  width: 80vw;
  max-width: 33rem;
  padding: 1rem;
  border: 2px solid hsla(0, 0%, 0%, 0.8);
  ${({ theme }) => css`
    @media ${theme.breakpoints.large} {
      width: 60vw;
    }
  `}
`;

const CloseButton = styled.button`
  border: none;
  background: transparent;
  width: 1.2rem;
  float: right;
  cursor: pointer;
`;

const Modal = ({ isOpen, onDismiss, children }) => {
  const AnimatedDialogOverlay = animated(DialogOverlayComponent);
  const AnimatedDialogContent = animated(DialogContentComponent);
  const transitions = useTransition(isOpen, {
    from: { opacity: 0, y: -10 },
    enter: { opacity: 1, y: 0 },
    leave: { opacity: 0, y: 10 },
  });
  return (
    <div>
      {transitions(
        (styles, item) =>
          item && (
            <AnimatedDialogOverlay style={{ opacity: styles.opacity }} onClick={onDismiss}>
              <AnimatedDialogContent
                style={{
                  transform: styles.y.to((value) => `translate3d(0px, ${value}px, 0px)`),
                }}
              >
                <CloseButton type="button" onClick={onDismiss}>
                  <CloseSVG />
                </CloseButton>
                {children}
              </AnimatedDialogContent>
            </AnimatedDialogOverlay>
          )
      )}
    </div>
  );
};

Modal.propTypes = {
  children: PropTypes.any,
  isOpen: PropTypes.bool,
  onDismiss: PropTypes.func,
};

export default Modal;
