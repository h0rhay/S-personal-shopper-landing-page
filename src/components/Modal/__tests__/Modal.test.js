import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { mockTheme } from '../../../../jest_config/mock';
import Modal from '../index';

const props = {
  isOpen: true,
  onDismiss: jest.fn(),
  children: <div>Modal Child Component</div>,
};

const renderComponent = ({ theme }) =>
  render(
    <ThemeProvider theme={theme}>
      <Modal {...props} />
    </ThemeProvider>
  );

describe('Modal component', () => {
  beforeEach(() => {
    renderComponent({ theme: mockTheme, props });
  });

  it('Close button should be in document', () => {
    expect(screen.getByTestId('close-button')).toBeInTheDocument();
    expect(screen.getByTestId('close-button')).toMatchSnapshot('CLOSE BUTTON');
  });

  it('Mock Child component should be in document', () => {
    expect(screen.getByText(/modal child component/i)).toBeInTheDocument();
    expect(screen.getByText(/modal child component/i)).toMatchSnapshot('MODAL CHILDREN COMPONENT');
  });
});
