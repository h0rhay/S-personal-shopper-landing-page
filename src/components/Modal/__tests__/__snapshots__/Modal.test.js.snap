// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Close button should be in document: CLOSE BUTTON 1`] = `
<button
  class="sc-dkPtRN fcQwMD"
  data-testid="close-button"
  type="button"
>
  <test-file-stub />
</button>
`;

exports[`Modal component Mock Child component should be in document: MODAL CHILDREN COMPONENT 1`] = `
<div>
  Modal Child Component
</div>
`;
